grails-app/controllers/org/bbop/apollo/AvailableStatusController.groovy

```diff
@@ -110,4 +112,177 @@ class AvailableStatusController {
+            , @RestApiParam(name = "old_name", type = "string", paramType = RestApiParamType.QUERY, description = "Status name to update")
+            , @RestApiParam(name = "new_name", type = "string", paramType = RestApiParamType.QUERY, description = "Status name to change to (the only editable option)")
+    ]
+    )
+    @Transactional
+    def updateStatus() {
+        try {
+            JSONObject statusJson = permissionService.handleInput(request, params)
+            log.debug "Updating status ${statusJson}"
+            if (permissionService.isUserAdmin(permissionService.getCurrentUser(statusJson))) {
+
+                log.debug "status ID: ${statusJson.id}"
+                AvailableStatus status = AvailableStatus.findById(statusJson.id)
+                if (!status) {
+                    status = AvailableStatus.findByLabel(statusJson.old_name)
+                }
+
+                if (!status) {
+                    JSONObject jsonObject = new JSONObject()
+                    jsonObject.put(FeatureStringEnum.ERROR.value, "Failed to update the status")
+                    render jsonObject as JSON
+                    return
+                }
+
+                status.label = statusJson.new_name
+                status.save(flush: true)
+
+                log.info "Success updating status: ${status.id}"
+                render new JSONObject() as JSON
+            } else {
+                def error = [error: 'not authorized to delete status']
+                log.error(error.error)
+                render error as JSON
+            }
+        }
+        catch (Exception e) {
+            def error = [error: 'problem deleting status: ' + e]
+            log.error(error.error)
+            render error as JSON
+        }
+    }
+
+    @RestApiMethod(description = "Remove a status", path = "/availableStatus/deleteStatus", verb = RestApiVerb.POST)
+    @RestApiParams(params = [
+            @RestApiParam(name = "username", type = "email", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "id", type = "long", paramType = RestApiParamType.QUERY, description = "Status ID to remove (or specify the name)")
+            , @RestApiParam(name = "name", type = "string", paramType = RestApiParamType.QUERY, description = "Status name to delete")
+    ])
+    @Transactional
+    def deleteStatus() {
+        try {
+            JSONObject statusJson = permissionService.handleInput(request, params)
+            log.debug "Deleting status ${statusJson}"
+            if (permissionService.isUserAdmin(permissionService.getCurrentUser(statusJson))) {
+
+                AvailableStatus status = AvailableStatus.findById(statusJson.id)
+                if (!status) {
+                    status = AvailableStatus.findByLabel(statusJson.name)
+                }
```

Comment:
see note above.  You can also put this in a private method if you end up re-using it.